compiler/modules/front/expr.bal

```diff
@@ -442,7 +442,8 @@ function codeGenMemberAccessExpr(ExprContext cx, bir:BasicBlock block1, Position
             var { result: r, block: nextBlock } = check codeGenExprForInt(cx, block1, index);
             t:SemType memberType = t:listMemberType(cx.mod.tc, l.semType, r.semType);
             if t:isEmpty(cx.mod.tc, memberType) {
-                return cx.semanticErr("type of member access is never", pos);
+                return t:listAtomicTypeRw(cx.mod.tc, l.semType) is () ? 
```

Comment:
Don't understand why you are checking whether l.semType is atomic.

If we are going to give two error messages here, I would distinguish the case where r.semType is a singleton.

Also use a conditional expression to compute the message, and then pass that to cx.semanticErr

Reply:
The original error is thrown for [09-bug/array3-e.bal](https://github.com/ballerina-platform/nballerina/blob/main/compiler/testSuite/09-bug/array3-e.bal), which can be distinguished from fixed length array by `t:listAtomicTypeRw`. Is there a better way to do this?